### PR TITLE
[Feature][SeaTunnel Engine] Add test to test stream job recover when all cluster node done and restart

### PR DESCRIPTION
**Pull request by TaoZex**
## Purpose of this pull request

https://github.com/apache/incubator-seatunnel/issues/3430
Add test to test stream job recover when all cluster node done and restart

## Check list

* [ ] Code changed are covered with tests, or it does not need tests for reason:
* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update change log that in connector document. For more details you can refer to [connector-v2](https://github.com/apache/incubator-seatunnel/tree/dev/docs/en/connector-v2)
  2. Update [plugin-mapping.properties](https://github.com/apache/incubator-seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  3. Update the pom file of [seatunnel-dist](https://github.com/apache/incubator-seatunnel/blob/dev/seatunnel-dist/pom.xml)

### CODE REVIEW

File: seatunnel-e2e/seatunnel-engine-e2e/connector-seatunnel-e2e-base/src/test/java/org/apache/seatunnel/engine/e2e/ClusterFaultToleranceIT.java [hunk 1]
```diff
@@ -611,4 +611,116 @@ public void testStreamJobRestoreIn3NodeMasterDown() throws ExecutionException, I
...
+            node1 = SeaTunnelServerStarter.createHazelcastInstance(
+                    TestUtils.getClusterName(testClusterName));
+
+            node2 = SeaTunnelServerStarter.createHazelcastInstance(
+                    TestUtils.getClusterName(testClusterName));
+
+            node3 = SeaTunnelServerStarter.createHazelcastInstance(
+                    TestUtils.getClusterName(testClusterName));
+
+            Thread.sleep(360000);
```
Comment (EricJoy2048) [inline seatunnel-e2e/seatunnel-engine-e2e/connector-seatunnel-e2e-base/src/test/java/org/apache/seatunnel/engine/e2e/ClusterFaultToleranceIT.java]: Why sleep 360000? I think we can wait all nodes added to cluster here.

**Review by EricJoy2048 — COMMENTED** [review]
Please merge the dev branch code.

File: seatunnel-e2e/seatunnel-engine-e2e/connector-seatunnel-e2e-base/src/test/java/org/apache/seatunnel/engine/e2e/ClusterFaultToleranceIT.java (hunk 1 above)
Comment (TaoZex) [reply seatunnel-e2e/seatunnel-engine-e2e/connector-seatunnel-e2e-base/src/test/java/org/apache/seatunnel/engine/e2e/ClusterFaultToleranceIT.java]: I have modified the code, but I cannot pass the CI. Is the code I modified wrong?

Comment (EricJoy2048) [reply seatunnel-e2e/seatunnel-engine-e2e/connector-seatunnel-e2e-base/src/test/java/org/apache/seatunnel/engine/e2e/ClusterFaultToleranceIT.java]: > I have modified the code, but I cannot pass the CI. Is the code I modified wrong?

Please wait https://github.com/apache/incubator-seatunnel/pull/3418 merge. And then you need config to enable `IMap storage` for this test case.

Comment (TaoZex) [reply seatunnel-e2e/seatunnel-engine-e2e/connector-seatunnel-e2e-base/src/test/java/org/apache/seatunnel/engine/e2e/ClusterFaultToleranceIT.java]: Thanks for your advice.

**Review by liugddx — APPROVED** [review]

**Review by EricJoy2048 — APPROVED** [review]

**Pull request merged**